src/Geometry/globalgeometry.jl

```diff
@@ -146,6 +146,66 @@ function great_circle_distance(
     )
 end
 
+"""
+    euclidean_distance(pt1::XYPoint, pt2::XYPoint)
+
+Compute the 2D Euclidean distance between `pt1` and `pt2`.
+"""
+function euclidean_distance(
+    pt1::XYPoint{FT},
+    pt2::XYPoint{FT},
+) where {FT}
+    return hypot((components(pt1) .- components(pt2))...)
+end
```

Comment:
You can avoid duplicate definitions by defining a single method and use "diagonal" dispatch:
```suggestion
function euclidean_distance(
    pt1::T,
    pt2::T,
) where {T<:Union{XYPoint,XZPoint,XYZPoint}
    return hypot((components(pt1) .- components(pt2))...)
end
```

Reply:
This way, it would only allow for same type. For hybrid spaces, wouldn't we want to allow a distance between two different types? Am I seeing this wrong?

Reply:
why would a hybrid space be different?